### PR TITLE
NBP modifications for host changes

### PR DESCRIPTION
**Pull request by sushanthakumar**
**What this PR does / why we need it**:
This PR is to address host management change impacts at NBP.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #
With host management changes, this also resolve the below issue
https://github.com/opensds/opensds/issues/1023

Test Report 
Except operations through helm chart, All test scenarios are executes  which includes
> Installation/uninstallation through installer and deployment scripts
> Creation deletion of pods with block volume
> Creation deletion of pods with fileshare

[csi-hostname-changes.tar.gz](https://github.com/opensds/nbp/files/3926282/csi-hostname-changes.tar.gz)

**Special notes for your reviewer**:
> CSI changes are completed

> Flex volume is not modified as support will not be continued, so build is disabled in makefile

> Cinder API changes for hostname impact is not merged with this as impact test not completed, will be handled seperately, cinder compilation is disabled in makefile

**Release note**:

```release-note
```

### CODE REVIEW

**Comment by sushanthakumar** [comment]
https://github.com/opensds/nbp/issues/295

File: util/util.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+package util
+
+import (
+	"errors"
+	"fmt"
+	"github.com/opensds/opensds/client"
+	"github.com/opensds/opensds/pkg/model"
+)
+
+// GetHostIdByHostName returns hostid for given hostname if found
```
Comment (NajmudheenCT) [inline util/util.go]: Comment has a different function name .

File: util/util.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+	for _, elem := range hostLists {
+		if elem.HostName == hostName {
+			return elem, nil
+		}
+	}
+	return nil, fmt.Errorf("Failed to find host for hostname %s", hostName)
+}
+
+// GetHostIdByHostName returns hostid for given hostId if found
```
Comment (NajmudheenCT) [inline util/util.go]: same here

File: util/util.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+)
+
+// GetHostIdByHostName returns hostid for given hostname if found
+func GetHostByHostName(client *client.Client, hostName string) (*model.HostSpec, error) {
+	hostLists, err := client.HostMgr.ListHosts()
+	if nil != err {
+		return nil, errors.New("Failed to list hosts")
+	}
+
+	for _, elem := range hostLists {
```
Comment (NajmudheenCT) [inline util/util.go]: Do we need to parse through all host list, Hope the API takes filter "?name=<hostName>"

File: util/util.go (hunk 1 above)
Comment (sushanthakumar) [reply util/util.go]: Resolved

File: util/util.go (hunk 2 above)
Comment (sushanthakumar) [reply util/util.go]: Resolved

File: util/util.go (hunk 3 above)
Comment (sushanthakumar) [reply util/util.go]: To be fixed. Raised an issue to track (https://github.com/opensds/nbp/issues/296)

File: csi/plugins/block/volume.go [hunk 4]
```diff
@@ -690,29 +631,12 @@ func (v *Volume) NodeUnstageVolume(req *csi.NodeUnstageVolumeRequest) (*csi.Node
...
 	return &csi.NodeUnstageVolumeResponse{}, nil
 }
 
 // NodePublishVolume implementation
 func (v *Volume) NodePublishVolume(req *csi.NodePublishVolumeRequest) (*csi.NodePublishVolumeResponse, error) {
-	volId := req.VolumeId
-	attachmentId := req.PublishContext[common.PublishAttachId]
-
-	if r := v.getReplicationByVolume(volId); r != nil {
-		volId = r.Metadata[AttachedVolumeId]
```
Comment (wisererik) [inline csi/plugins/block/volume.go]: why delete replication code here?

File: csi/common/node.go [hunk 5]
```diff
@@ -119,38 +123,60 @@ func NodeGetInfo(
...
+		}
 
-	for _, volDriverType := range volDriverTypes {
-		volDriver := connector.NewConnector(volDriverType)
-		if volDriver == nil {
-			glog.Errorf("unsupport volume driver: %s", volDriverType)
-			continue
+		hostSpec := &model.HostSpec{
+			TenantId:          TenantId,
+			HostName:          hostName,
```
[the remark this answers is not in the document]
Comment (sushanthakumar) [reply csi/common/node.go]: Resolved

File: csi/plugins/block/volume.go [hunk 6]
```diff
@@ -262,62 +262,9 @@ func (v *Volume) ControllerPublishVolume(req *csi.ControllerPublishVolumeRequest
...
-			Host:      nodeInfoValues[0],
-			Platform:  runtime.GOARCH,
-			OsType:    runtime.GOOS,
-			Initiator: initator,
-			Ip:        nodeInfoValues[len(nodeInfoValues)-1],
-		},
-		Metadata:       req.VolumeContext,
+		VolumeId:       req.VolumeId,
+		HostId:         req.NodeId,
 		AccessProtocol: protocol,
```
[the remark this answers is not in the document]
Comment (sushanthakumar) [reply csi/plugins/block/volume.go]: Resolved

File: csi/common/node.go [hunk 7]
```diff
@@ -119,38 +123,60 @@ func NodeGetInfo(
...
+				PortName: portName,
+				Protocol: volDriverType,
+			}
+
+			initiators = append(initiators, initiator)
+		}
 
-	volDriverTypes := []string{connector.FcDriver, connector.IscsiDriver, connector.NvmeofDriver}
+		if len(initiators) == 0 {
+			msg := fmt.Sprintf("cannot get any initiator for host %s", hostName)
```
[the remark this answers is not in the document]
Comment (sushanthakumar) [reply csi/common/node.go]: Resolved

File: csi/plugins/block/volume.go (hunk 4 above)
Comment (sushanthakumar) [reply csi/plugins/block/volume.go]: Resolved

**Pull request merged**